### PR TITLE
Use cml to create release PR

### PR DESCRIPTION
**Pull request by dacbd**
Part of an effort to use our own tools more/where appropriate.

This current state will replace what was present. However, `cml pr create` doesn't have an option for labels, so some additional bash script is required.

`cml pr create` output prints the pr URL like `https://github.com/iterative/vscode-dvc/pull/2998` and `awk -F / '{print $NF}'` splits on `/` and prints the last field being the pr # for gh to add the label.

---

cc @iterative/vs-code feel free to merge as is if this is acceptable to you, but I think its fine to hold out for a cml feature to support labels

---
cc @casperdcl I think a `--labels` option is an easy addition.

### CODE REVIEW

File: .github/workflows/create-release-pr.yml [hunk 1]
```diff
@@ -51,16 +54,19 @@ jobs:
...
+          echo "**The \`publish\` workflow will run automatically when this PR is merged.**" >> pr-body.md
 
-            Please close and reopen this PR to run the required workflows (Required statuses must pass before merging).
+          new_pr=$(cml pr create \
+            --token="${{ secrets.GITHUB_TOKEN }}" \
+            --message="Update version and CHANGELOG for release" \
+            --title="Update version and CHANGELOG for release" \
+            --body="$(cat ./pr-body.md)")
+          
+          gh pr edit $(echo "$new_pr" | awk -F / '{print $NF}') --add-label "make release go now"
```
Comment (mattseddon) [inline .github/workflows/create-release-pr.yml:70]: @dacbd the label isn't important, we just need to drop it from the if statement in `.github/workflows/publish.yml`

**Comment by dacbd** [comment]
I'll take a look and push some additional changes.

**Comment by mattseddon** [comment]
Can we use CML to create a PR in this repository every time our sub-repo has been updated (https://github.com/iterative/vscode-dvc-demo)?

**Comment by dacbd** [comment]
> Can we use CML to create a PR in this repository every time our sub-repo has been updated (https://github.com/iterative/vscode-dvc-demo)?

@mattseddon I think we can definitely do this. The pattern would be for `vscode-dvc-demo` on push to main, trigger a `repositiory_dispatch` here which would update the submodule and create a PR. I can do this in a separate set or PRs just make an issue an assign it to me and I'll likely get it done before next monday.

**Review by mattseddon — APPROVED** [review]

**Comment by mattseddon** [comment]
> > Can we use CML to create a PR in this repository every time our sub-repo has been updated (https://github.com/iterative/vscode-dvc-demo)?
> 
> @mattseddon I think we can definitely do this. The pattern would be for `vscode-dvc-demo` on push to main, trigger a `repositiory_dispatch` here which would update the submodule and create a PR. I can do this in a separate set or PRs, just make an issue an assign it to me and I'll likely get it done before next monday.

Will leave it with you 🙏🏻

**Pull request merged**